internal/dataplane/parser/translators/ingress.go

```diff
@@ -98,7 +98,7 @@ func (i *ingressTranslationIndex) add(ingress *netv1.Ingress) {
 			serviceName := httpIngressPath.Backend.Service.Name
 			servicePort := httpIngressPath.Backend.Service.Port.Number
 
-			cacheKey := fmt.Sprintf("%s%s%s%s%d", ingress.Namespace, ingress.Name, ingressRule.Host, serviceName, servicePort)
+			cacheKey := fmt.Sprintf("%s.%s.%s.%s.%d", ingress.Namespace, ingress.Name, ingressRule.Host, serviceName, servicePort)
```

Comment:
This alone prevents from overriding rules that have same substring `<ingressRule.Host><serviceName>` or more likely `<serviceName><servicePort>`